src/wslbridge2.cpp

```diff
@@ -384,12 +385,12 @@ int main(int argc, char *argv[])
     {
         // wsltty#302: Start dummy process after ComInit, otherwise RPC_E_TOO_LATE.
         // wslbridge2#38: Do this only for WSL2 as WSL1 does not need the VM context.
+        // wslbridge2#42: Required for WSL2 to get the VM ID.
         if (LiftedWSLVersion)
             start_dummy(wslPath, wslCmdLine, distroName, debugMode);
 
-        const HRESULT hRes = GetVmId(&DistroId, &VmId, LiftedWSLVersion);
```

Comment:
Instead of removing the COM logic, would it be possible to add the registry check after COM failed? 

After this line 

https://github.com/Biswa96/wslbridge2/blob/5b2b652d1a7355b004e7860b4370a585737e5ac9/src/GetVmId.cpp#L312

Reply:
done